GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -182,8 +182,17 @@ private bool TryInitialize(out string errorMessage)
         {
             if (this.upgrader == null)
             {
+                string gitBinPath = GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath();
+                if (string.IsNullOrEmpty(gitBinPath))
+                {
+                    errorMessage = $"nameof(this.TryInitialize): Unable to locate git installation. Ensure git is installed and try again.";
+                    return false;
+                }
+
+                ICredentialStore credentialStore = new GitProcess(gitBinPath, workingDirectoryRoot: null, gvfsHooksRoot: null);
+
                 ProductUpgrader upgrader;
-                if (!ProductUpgrader.TryCreateUpgrader(this.tracer, this.fileSystem, this.DryRun, this.NoVerify, out upgrader, out errorMessage))
+                if (!ProductUpgrader.TryCreateUpgrader(this.tracer, this.fileSystem, new LocalGVFSConfig(), credentialStore, this.DryRun, this.NoVerify, out upgrader, out errorMessage))

```

Comment:
Is there a way we can share this code and use it here and in the UpgradeVerb?

Reply:
Probably :) I hadn't looked at it too closely, but I suspect the answer is yes.... but outside the scope of this PR.